### PR TITLE
feat(tracing): add support for flame graph generation

### PR DESCRIPTION
**Pull request by rvolosatovs**
## Feature or Problem

Generate flame graphs using https://docs.rs/tracing-flame/latest/tracing_flame/index.html if configured to do so, also refactor the observability function to do that in a sane way

### CODE REVIEW

File: crates/tracing/src/lib.rs [hunk 1]
```diff
@@ -6,6 +6,7 @@ pub use opentelemetry::{
     metrics::{Counter, Histogram, Meter, Unit},
     KeyValue,
 };
+use std::path::Path;
```
Comment (thomastaylor312) [inline crates/tracing/src/lib.rs:9]: Super nit: Should this import go up top since it is `std`?

**Review by thomastaylor312 — APPROVED** [review]

**Comment by brooksmtownsend** [comment]
Going to give this a shot tomorrow with our tracing infra 😄

File: crates/tracing/src/lib.rs (hunk 1 above)
Comment (thomastaylor312) [reply crates/tracing/src/lib.rs:9]: Resolving this so it can merge since this is just a super nit

Comment (rvolosatovs) [reply crates/tracing/src/lib.rs:9]: I really was not sure due to all the feature-gated imports. I think normally you'd define a feature-gated module instead, where all imports are just imports (no features)

**Pull request merged**